src/java/com/sarality/app/view/dialog/CustomViewDialogFragment.java

```diff
@@ -85,6 +85,8 @@ protected void reinitialize() {
    * @param loaderId Id to used for the loader
    */
   protected <D> void startLoad(DataSource<D> dataSource, int loaderId, DataConsumer<D> consumer) {
+    // Kill all previously loaded data by the activity
+    getActivity().getSupportLoaderManager().destroyLoader(loaderId);
```

Comment:
I assume different loaderIds were being used for these calls. In which case, the null data store should be returning null on each call. The previous AsyncTask call shouldn't matter since it has a different loaderId.

Let's sit dow together next week and resolve all this loader design. I'd like this to be resolved once and for all. Killing a previous loader has lot of other problems associated with it.